src/engine/Server.cpp

```diff
@@ -611,7 +612,8 @@ boost::asio::awaitable<void> Server::processQuery(
               << (pinResult ? " [pin result]" : "")
               << (pinSubtrees ? " [pin subresults]" : "") << "\n"
               << query << std::endl;
-    ParsedQuery pq = SparqlParser::parseQuery(query);
+    ParsedQuery pq = co_await computeInNewThread(
+        [&query]() { return SparqlParser::parseQuery(query); });
```

Comment:
I checked, the variable `pq` is not used again until the next call to `computeInNewThread`, 
So you might as well merge those two calls (the parsing and the query planning) into a single `computeInNewThread` and extract that block to a function, s.t. we can simply write something like
`auto [pq, qet] = co_await parseAndPlan(query, ...)` as this function already is way too long.